### ISSUE TITLE
A way to stop error message spam when no internet

### ISSUE THREAD

**Issue by RobertSkalko** [gorilla-devs/GDLauncher-Carbon#359]
A few weeks ago I had an internet outage, so as any sane person, I wanted to play my singleplayer games like.. minecraft

So I opened up gdlauncher and was met with 10 different errors.

The errors blocked the modpack button so I needed to tap away for half a minute before being able to launch a modpack

Errors pop back up after being closed

Can the errors either be in a place they don't mess with the gui or stop coming if i click the x button on them?

Specifically the launcher shows a bunch of "request error" on the left side

This isn't a massive problem, just annoying enough that I remembered it weeks later

**Comment by D-Raj-Chakraborty** [comment]
#Solution: GDLauncher Error Behavior During Internet Outage
---

## Proposed Solution

### 1. **Introduce an Offline Mode**
   - Add a dedicated offline mode that automatically activates when no internet connection is detected.
   - Behavior of offline mode:
     - Suppress network-related error messages.
     - Disable features requiring internet access (e.g., modpack downloads, updates).
     - Allow full access to local modpacks and single-player games.

   **Implementation Steps**:
   - Detect internet connectivity at application startup using a lightweight ping or connection test.
   - If no connection is found, switch to offline mode and display an unobtrusive banner (e.g., "Offline Mode: Some features are unavailable").

---

### 2. **Error Message Redesign**
   - Ensure error messages do not obstruct core GUI elements like the modpack button.
   - Suggested changes:
     - Display errors in a dedicated **notification panel** on the side or bottom of the launcher.
     - Provide a collapsible or dismissible error area to reduce user annoyance.
     - Avoid modal popups for repetitive errors during offline use.

   **Mock Behavior**:
   - Errors such as "request error" appear in a non-blocking toast notification or as a line in the notification panel.
   - Allow users to clear all notifications in one click.

---

### 3. **Dismissal Persistence**
   - Ensure that once the user dismisses an error message using the **X** button, it does not reappear unless:
     - A new error occurs.
     - The user refreshes or restarts the application.

   **Technical Approach**:
   - Store the dismissed state of errors locally (e.g., in a cache or temporary file).
   - Reset error notifications only if the connection is restored or the issue changes.

---

### 4. **Testing and Validation**
   - Simulate various scenarios to validate the fix:
     - Launch GDLauncher offline and ensure no blocking errors appear.
     - Verify that local modpacks and single-player games are fully accessible.
     - Test how error messages behave when transitioning between offline and online states.

**Comment by blarfoon** [comment]
Yeah I understand it can be very annoying. I'm gonna fix it. Thank you for reporting it

**Comment by blarfoon** [comment]
Fixed in the next release: connection errors now collapse into a single notification instead of a popup flood, and identical errors are deduplicated.
